### PR TITLE
[ADD/IMP][13.0] Added and Improved test cases.

### PR DESCRIPTION
**Pull request by rutul18**

### CODE REVIEW

**Comment by JayVora-SerpentCS** [comment]
@YogendrasinhChavda can you review please?
Thanks.

**Comment by YogendrasinhChavda** [comment]
Okay sir, i will review.

File: fleet_rent/tests/test_rent_type.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+class TestRentType(TransactionCase):
+
+    def setUp(self):
+        super(TestRentType, self).setUp()
+
+    def test(self):
+        self.rent_type_obj = self.env['rent.type']
+
```
Comment (YogendrasinhChavda) [inline fleet_rent/tests/test_rent_type.py]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_tag.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+class TestVehicleTag(TransactionCase):
+
+    def setUp(self):
+        super(TestVehicleTag, self).setUp()
+
+    def test(self):
+        vehicle_tag_obj = self.env['fleet.vehicle.tag']
+       
+        vehicle_tag = vehicle_tag_obj.search([('name','=','Break')])
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_tag.py]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_tag.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+class TestVehicleTag(TransactionCase):
+
+    def setUp(self):
+        super(TestVehicleTag, self).setUp()
+
+    def test(self):
+        vehicle_tag_obj = self.env['fleet.vehicle.tag']
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_tag.py]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_registration.py [hunk 4]
```diff
@@ -13,10 +12,34 @@ def test(self):
...
+        vehicle_registration_state = registration_state_obj.search([('name','=', 'Gujarat')])            
+       
+        vehicle_color = vehicle_color_obj.search([('name','=','BLACK')])
+
+        vehicle_type = vehicle_type_obj.search([('name','=','Car')])
+                                       
+        self.driver = driver_obj.create({
+                            'name':'Shivam',
+                            'email':'shivam@gmail.com'})
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_registration.py]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_registration.py [hunk 5]
```diff
@@ -13,10 +12,34 @@ def test(self):
...
+        driver_obj = self.env['res.partner']
+        cr_dt = datetime.today().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+
+
+        vehicle_registration_state = registration_state_obj.search([('name','=', 'Gujarat')])            
+       
+        vehicle_color = vehicle_color_obj.search([('name','=','BLACK')])
+
+        vehicle_type = vehicle_type_obj.search([('name','=','Car')])
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_registration.py:26]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_registration.py [hunk 6]
```diff
@@ -13,10 +12,34 @@ def test(self):
...
+        vehicle_color_obj = self.env['color.color']
+        registration_state_obj = self.env['res.country.state']
+        driver_obj = self.env['res.partner']
+        cr_dt = datetime.today().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+
+
+        vehicle_registration_state = registration_state_obj.search([('name','=', 'Gujarat')])            
+       
+        vehicle_color = vehicle_color_obj.search([('name','=','BLACK')])
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_registration.py]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_registration.py [hunk 7]
```diff
@@ -13,10 +12,34 @@ def test(self):
...
         fleet_model = self.env.ref('fleet.model_a1')
+        vehicle_type_obj = self.env['vehicle.type']
+        vehicle_color_obj = self.env['color.color']
+        registration_state_obj = self.env['res.country.state']
+        driver_obj = self.env['res.partner']
+        cr_dt = datetime.today().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+
+
+        vehicle_registration_state = registration_state_obj.search([('name','=', 'Gujarat')])            
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_registration.py:23]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_registration.py [hunk 8]
```diff
@@ -13,10 +12,34 @@ def test(self):
         self.vehicle_obj = self.env['fleet.vehicle']
         fleet_brand = self.env.ref('fleet.brand_audi')
         fleet_model = self.env.ref('fleet.model_a1')
+        vehicle_type_obj = self.env['vehicle.type']
+        vehicle_color_obj = self.env['color.color']
+        registration_state_obj = self.env['res.country.state']
+        driver_obj = self.env['res.partner']
+        cr_dt = datetime.today().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_registration.py:21]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_odometer.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+        odometer_obj = self.env['fleet.vehicle.odometer']
+        fleet_vehicle = self.env.ref('fleet.vehicle_1')
+        cr_dt = datetime.today().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+        driver_obj = self.env['res.partner']
+
+        self.driver = driver_obj.create({
+                            'name':'Shivam',
+                            'email':'shivam@gmail.com',
+                            'phone': 9898981213})
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_odometer.py:21]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_odometer.py [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    def setUp(self):
+        super(TestVehicleOdometer, self).setUp()
+
+    def test(self):
+        odometer_obj = self.env['fleet.vehicle.odometer']
+        fleet_vehicle = self.env.ref('fleet.vehicle_1')
+        cr_dt = datetime.today().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+        driver_obj = self.env['res.partner']
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_odometer.py:16]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_model.py [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+class TestVehicleModel(TransactionCase):
+
+    def setUp(self):
+        super(TestVehicleModel, self).setUp()
+
+    def test(self):
+        self.vehicle_obj = self.env['fleet.vehicle.model']
+        fleet_brand = self.env['fleet.vehicle.model.brand'].create({
+                                                    'name':'i20'})
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_model.py:13]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_fuel.py [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+class TestVehicleFuel(TransactionCase):
+
+    def setUp(self):
+        super(TestVehicleFuel, self).setUp()
+
+    def test(self):
+        fuel_obj = self.env['fleet.vehicle.log.fuel']
+        fleet_vehicle = self.env.ref('fleet.vehicle_1')
+        cr_dt = datetime.today().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_fuel.py:15]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_driver.py [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+        super(TestVehicleDriver, self).setUp()
+
+    def test(self):
+        driver_obj = self.env['res.partner']
+        self.driver = driver_obj.create({
+                                'name':'Shivam',
+                                'email':'shivam@gmail.com',
+                                'phone': 9898981213})
+        
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_driver.py:16]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_cost.py [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+
+    def test(self):
+        cost_obj = self.env['fleet.vehicle.cost']
+        fleet_vehicle = self.env.ref('fleet.vehicle_1')
+        service_type_obj = self.env['fleet.service.type']
+        cr_dt = datetime.today().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+
+        self.vehicle_service = service_type_obj.create({
+                                'name':'Washing'})
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_cost.py]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_cost.py [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+    def setUp(self):
+        super(TestVehicleCost, self).setUp()
+
+    def test(self):
+        cost_obj = self.env['fleet.vehicle.cost']
+        fleet_vehicle = self.env.ref('fleet.vehicle_1')
+        service_type_obj = self.env['fleet.service.type']
+        cr_dt = datetime.today().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_cost.py]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_color.py [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+
+class TestVehicleColor(TransactionCase):
+
+    def setUp(self):
+        super(TestVehicleColor, self).setUp()
+
+    def test(self):
+        vehicle_color_obj = self.env['color.color']
+       
+        vehicle_color = vehicle_color_obj.search([('name','=','BLACK')])
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_color.py]: add limit=1 in search

File: fleet_operations/tests/test_next_service_day.py [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+
+class TestNextServiceDay(TransactionCase):
+
+    def setUp(self):
+        super(TestNextServiceDay, self).setUp()
+
+    def test(self):
+        next_service_obj = self.env['next.service.days']
+        fleet_vehicle = self.env.ref('fleet.vehicle_1')
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_next_service_day.py]: Remove Extra line.

File: fleet_operations/tests/test_vehicle_color.py [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+
+
+class TestVehicleColor(TransactionCase):
+
+    def setUp(self):
+        super(TestVehicleColor, self).setUp()
+
+    def test(self):
+        vehicle_color_obj = self.env['color.color']
+
```
Comment (YogendrasinhChavda) [inline fleet_operations/tests/test_vehicle_color.py]: Remove Extra line.

File: fleet_rent/tests/test_rent_type.py [hunk 19]
```diff
@@ -0,0 +1,15 @@
+from odoo.tests.common import TransactionCase
+
+
+class TestRentType(TransactionCase):
+
+    def setUp(self):
+        super(TestRentType, self).setUp()
+
+    def test(self):
+        self.rent_type_obj = self.env['rent.type']
```
Comment (YogendrasinhChavda) [inline fleet_rent/tests/test_rent_type.py]: No need extra global variable if you are using only one time object to do the operation ( create, write .. etc)
you can directly use  self.env['rent.type'].create

**Pull request merged**